README.md

```diff
@@ -14,42 +14,114 @@ This allows implementers of the Reactive Socket protocol to be able to write the
 
 The DSLs are designed to be human readable as well, and should require very little documentation to understand. Here is an example of the client side DSL
 ```
-val s1 = requestResponse("a", "b")
-    s1 request 1
-    val s2 = requestResponse("c", "d")
+  def main(args: Array[String]) {
+    begintest(test0)
+    begintest(test4)
+    end
+  }
+
+  // example for testing channel
+  def test0() : Unit = {
+    // using determines the initial payload sent
+    requestChannel using("a", "b") asFollows(() => {
+      respond("-a-")
+      val s1 = channelSubscriber
+      s1 request 1
+      respond("-b-c-d-e-f-")
+      s1 awaitAtLeast(1, 2000)
+      s1 assertReceivedAtLeast 1
+      s1 assertReceived List(("x", "x"))
+      s1 request 2
+      respond("-g-h-i-j-k-")
+      s1 awaitAtLeast(4, 2000)
+      s1 request 4
+      s1 awaitAtLeast(7, 1000)
+      respond("|")
+      s1 awaitTerminal()
+      s1 assertCompleted()
+    })
+  }
+
+....
+
+  // example for testing stream
+  def test4() : Unit = {
+    nametest("test4")
+    val s1 = requestStream("a", "b")
+    s1 request 3
+    val s2 = requestStream("c", "d")
+    s1 awaitAtLeast(3, 2000)
+    s2 request 1
+    s1 assertReceived(List(("a", "b"), ("c", "d"), ("e", "f")))
+    s1 request 3
     s1 awaitTerminal()
     s1 assertCompleted()
     s1 assertNoErrors()
-    s2 request 1
-    val s3 = requestResponse("e", "f")
-    s2 awaitTerminal()
-    s2 assertCompleted()
-    s3 request 1
+    s1 assertReceivedCount 6
+    s2 cancel()
+    s2 assertCanceled()
     s2 assertNoErrors()
-    s1 assertReceived List(("hello", "goodbye"))
-    s2 assertReceived List(("ding", "dong"))
-    s3 awaitTerminal()
-    s3 assertError()
+  }
 ```
 In the above example, we create a requestResponse subscriber and subscribe it to a trivial publisher with initial payload data "a" and metadata "b". We then call request on it, and assert the behaviors we expect.
 When we assert received, since it is a requestResponse, we only expect a list of 1 payload, and we assert both the data and metadata received. In this example, we are testing interleaving 3 requestResponse subscribers.
 
+Notice that the test for channel incorporates both server and client behavior. You are able to send data and assert data received. The IO is non-blocking, while the await blocks the thread running the main
+tests. So for example, calling `respond(...)` and then `request ...` will not block the request if the client can't respond, but calling `await ...` will block anything after it, but not respond requests
+that have already be started.
```

Comment:
Thanks :) , I will create better documentation in the future, and if you'd like any more documentation now, I'd be happy to provide it.